python/pyspark/sql/functions.py

```diff
@@ -349,6 +388,50 @@ def when(condition, value):
     return Column(jc)
 
 
+@since(1.4)
+def lag(col, count=1, default=None):
+    """
+    Window function: returns the lag values of current row of the expression,
+    given default value when the current row extends before the beginning
+    of the window.
+
+    :param col: name of column or expression
+    :param count: number of row to extend
+    :param default: default value
+    """
+    sc = SparkContext._active_spark_context
+    return Column(sc._jvm.functions.lag(_to_java_column(col), count, default))
+
+
+@since(1.4)
+def lead(col, count=1, default=None):
+    """
+    Window function: returns the lead values of current row of the column,
+    given default value when the current row extends before the end of the window.
+
+    :param col: name of column or expression
+    :param count: number of row to extend
+    :param default: default value
+    """
+    sc = SparkContext._active_spark_context
+    return Column(sc._jvm.functions.lead(_to_java_column(col), count, default))
+
+
+@since(1.4)
+def ntile(n):
+    """
+    NTILE for specified column or expression.
```

Comment:
I will submit a patch to udpate scaladoc for all scala methods. the docs are not there yet.